lib/hq/graphql/object.rb

```diff
@@ -80,6 +80,10 @@ def field_from_association(association, auto_nil:, internal_association: false,
               end
               instance_eval(&block) if block
             end
+          when :has_one
+            field name, type, null: null_from_field?(name) || !auto_nil || !association_required?(association), klass: model_name do
```

Comment:
has_one associations should only be nullable if there is a "presence" validation or we're overriding null with `auto_nil`